packages/flutter/lib/src/painting/matrix_utils.dart

```diff
@@ -169,6 +169,60 @@ class MatrixUtils {
     transform = new Matrix4.copy(transform)..invert();
     return transformRect(transform, rect);
   }
+
+  /// Create a transformation matrix which mimics the effects of tangentially
+  /// wrapping the plane on which this transform is applied around a cylinder
+  /// and then looking at the cylinder from a point outside the cylinder.
+  ///
+  /// [radius] simulates the radius of the cylinder the plane is being wrapped
+  /// onto. If the transformation is applied to a 0-dimensional dot instead of
+  /// a plane, the dot would simply translate by +/- [radius] pixels along the
+  /// [orientation] [Axis] when rotating from 0 to +/- 90 degrees.
+  ///
+  /// A positive radius means the object is closest at 0 [angle] and a negative
+  /// radius means the object is closest at [pi] [angle].
+  ///
+  /// [angle] is the difference in angle in radians between the object and the
+  /// viewing point. A positive [angle] on a positive [radius] moves the object
+  /// up when [orientation] is vertical and right when horizontal.
+  ///
+  /// The transformation is always done such that a 0 [angle] keeps the
+  /// transformed object at exactly the same size as before regardless of
+  /// [radius] and [perspective] when [radius] is positive.
+  ///
+  /// [perspective] is a number between 0 and 1 where 0 means looking at the
+  /// object from infinitely far with an infinitely narrow field of view and
+  /// 1 means looking at the object from infinitely close with an infinitely wide
+  /// field of view. Defaults to a sane but arbitrary 0.001.
+  ///
+  /// [orientation] is the direction of rotation of the cylinder.
+  static Matrix4 cylindricalProjectionTransform({
+    @required double radius,
+    @required double angle,
+    double perspective: 0.001,
```

Comment:
`assert` that `perspective` is between 0 and 1?

Reply:
done